### PR TITLE
Added status code check

### PR DESCRIPTION
**Pull request by stubbo**
lets not say success when it fails 🙄

### CODE REVIEW

File: lua/cosmo/core/sv_http.lua [hunk 1]
```diff
@@ -27,6 +27,28 @@ function HTTP_CLIENT:SetAuthorizationToken(token)
     self.Defaults.Headers["Authorization"] = "Bearer " .. token
 end
 
+local function httpSuccess(code, body, headers)
+    if code != 200 then
```
Comment (mexdeloo) [inline lua/cosmo/core/sv_http.lua]: Success doesn't always imply it's a 200, for example the API returns 204 aswell for the update endpoints

**Review by mexdeloo — CHANGES_REQUESTED** [review]

**Review by mexdeloo — CHANGES_REQUESTED** [review]
This PR would throw several errors.
Line 32, httpFailed will be nil, httpFailed needs to be defined before httpSuccess if you want to use it in httpSuccess. Due to it being a local function

Also `promise` is not passed as a parameter so would throw an error aswell.
Same goes for several other variables (url, reason)

File: lua/cosmo/core/sv_http.lua [hunk 2]
```diff
@@ -27,6 +27,28 @@ function HTTP_CLIENT:SetAuthorizationToken(token)
     self.Defaults.Headers["Authorization"] = "Bearer " .. token
 end
 
+local function httpSuccess(code, body, headers)
+    if code >= 400 then
+        return httpFailed("Returned invalid response code: " .. code)
+    end
+    
+    Cosmo.Log.Debug("(HTTP)", "Request succeeded, status code:", code)
+    Cosmo.Log.Debug("Endpoint:", url)
```
Comment (mexdeloo) [inline lua/cosmo/core/sv_http.lua]: `url` is not defined here

File: lua/cosmo/core/sv_http.lua [hunk 3]
```diff
@@ -27,6 +27,28 @@ function HTTP_CLIENT:SetAuthorizationToken(token)
...
+
+    if body and headers["Content-Type"] == "application/json" then
+        body = util.JSONToTable(body)
+    end
+
+    promise:Resolve(body, code, headers)
```
Comment (mexdeloo) [inline lua/cosmo/core/sv_http.lua]: `promise` is not defined here

File: lua/cosmo/core/sv_http.lua [hunk 4]
```diff
@@ -27,6 +27,28 @@ function HTTP_CLIENT:SetAuthorizationToken(token)
     self.Defaults.Headers["Authorization"] = "Bearer " .. token
 end
 
+local function httpSuccess(code, body, headers)
+    if code >= 400 then
+        return httpFailed("Returned invalid response code: " .. code)
```
Comment (mexdeloo) [inline lua/cosmo/core/sv_http.lua]: httpFailed is nil here

**Comment by mexdeloo** [comment]
Should be good to merge when these errors are fixed.

**Comment by Lythium4848** [comment]
fixed :)

**Review by FelixKLG — CHANGES_REQUESTED** [review]
add the term penis to every error message

**Pull request closed without merging**